src/cartesian/CartesianGrid.tsx

```diff
@@ -371,7 +398,12 @@ export function CartesianGrid(props: Props) {
     height: isNumber(props.height) ? props.height : offset.height,
   };
 
-  const { x, y, width, height, xAxis, yAxis, syncWithTicks, horizontalValues, verticalValues } = propsIncludingDefaults;
+  const { x, y, width, height, syncWithTicks, horizontalValues, verticalValues } = propsIncludingDefaults;
+
+  // @ts-expect-error the scale prop is mixed up - we need to untagle this at some point
```

Comment:
Hmm https://github.com/recharts/recharts/blob/master/src/util/types.ts#L26

We definitely aren't using correct scale types lol

Reply:
Yep, Recharts will accept `scale` externally as a string, and then parse it and evaluate as one of the d3 scales - but the types for the "external scale" and "internal scale" are mixed. It's on my list.